knowledge-base/upload-validateantiforgerytoken.md

```diff
@@ -0,0 +1,226 @@
+---
+title: Upload Files with Antiforgery Validation
+description: Learn how to setup the Telerik Blazor Upload to work with .NET controllers that use ValidateAntiForgeryToken attribute. Configure Blazor apps with antiforgery validation.
+type: how-to
+page_title: How to Upload Files with Antiforgery Validation
+slug: upload-kb-validateantiforgerytoken
+position: 
+tags: telerik, blazor, upload
+ticketid: 1626509, 1637325
+res_type: kb
+---
+
+## Environment
+
+<table>
+    <tbody>
+        <tr>
+            <td>Product</td>
+            <td>Upload for Blazor</td>
+        </tr>
+    </tbody>
+</table>
+
+
+## Description
+
+This KB article answers the following questions:
+
+* How to use the Telerik Blazor Upload component with controllers that are decorated with `[ValidateAntiForgeryToken]` attribute?
+* How to upload files to controllers that require antiforgery validation?
+* How to set antiforgery tokens in the Upload's `OnUpload` and `OnRemove` events?
+
+
+## Solution
+
+Here are the suggested steps to configure .NET Core Blazor antiforgery validation and integrate it with the Telerik Upload component.
+
+1. Edit `Program.cs`:
+    * Add `builder.Services.AddRazorPages();`
+    * Add `builder.Services.AddHttpContextAccessor();`
+    * (optional) Add `builder.Services.AddAntiforgery()` with custom `HeaderName` or `FormFieldName`
+    * Verify that `app.UseAntiforgery();` is present
+    * Add `app.MapDefaultControllerRoute();` to configure routing
+1. [Implement the `Save` and `Remove` controller methods]({%slug upload-overview%}#implement-controller-methods).
+1. Decorate the controler class or specific action methods with `[ValidateAntiForgeryToken]`.
+1. Configure the Razor component, which contains the Telerik Blazor Upload:
+    * Inject `AntiforgeryStateProvider` to use its `GetAntiforgeryToken()` method
```

Comment:
Consider clarifying the `its` possessive. Right now, it's not immediately clear what it stands for - is it the Razord `GetAntiforgeryToken` or the Blazor one?

```suggestion
    * Inject `AntiforgeryStateProvider` to use its `GetAntiforgeryToken()` method.
```

Reply:
Mentioning a dependency injection in combination with the same dependency's method or property should be intuitive to every developer. Those who get confused will rely on the code example below anyway.